cdn_lambda/functions/map_to_s3/map_to_s3.py

```diff
@@ -1,32 +1,36 @@
 import json
 import logging
-import os
 from datetime import datetime, timezone
 
 import boto3
 
 LOG = logging.getLogger("map-to-s3-lambda")
 
 
-def lambda_handler(event, context):
+def load_conf(conf_file):
+    with open(conf_file, "r") as json_file:
+        content = json.load(json_file)
+
+    return content
+
+
+def lambda_handler(event, context, conf_file=None):
     # pylint: disable=unused-argument
 
-    # This AWS Lambda function must be deployed with DB_TABLE_NAME and
-    # DB_TABLE_REGION environment variables
-    table_name = os.getenv("DB_TABLE_NAME")
-    table_region = os.getenv("DB_TABLE_REGION")
+    conf_file = conf_file or "map_to_s3.json"
+    conf = load_conf(conf_file)
```

Comment:
I don't think we should be doing this here.  Loading a JSON file is not all that expensive, but nevertheless doing it for every single request is unnecessarily wasteful and it'd be best to figure out how to handle one-time initialization like this.  (This is https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html point "Take advantage of execution context reuse to improve the performance of your function")

I think we should be doing something more like this:

```python
class PreparedLambda:
  def __init__(self, conf_file="map_to_s3.json"):
    # This code runs only when the package is initially imported
    self.conf = load_conf(conf_file)

  def lambda_handler(self, event, context):
    # ...lambda_handler body, as it is now; may refer to self.conf as needed

# Then make a 'lambda_handler' available at the module level which is bound to
# a particular instance of the class
lambda_handler = PreparedLambda().lambda_handler
```

That would result in the config file being loaded only once, when map_to_s3 is imported.  It doesn't compromise testability of the code, since tests can freely construct a PreparedLambda pointing at the desired config.

Reply:
I like this. Going to take one step further and initialize the boto3 client in the class as well.